### PR TITLE
EDUCATOR-4839: Team-enabled ORAs should not be actionable by teamless users

### PR DESCRIPTION
**Pull request by jansenk**
@edx/masters-devs 
[EDUCATOR-4839](https://openedx.atlassian.net/browse/EDUCATOR-4839)
Hide ORA and display message if the ora is team-based and the user is not on a team.
IF the user is course staff or if we are rendering the XBlock in studio preview, use some dummy values to render the page.

ALSO, refactored existing and new teams functionality into TeamsMixin.py and moved testing into test_teams.py

Learner not on a team:
Screen Shot 2019-12-20 at 11 58 42 AM

Course staff not on a team
Screen Shot 2019-12-20 at 11 01 21 AM

### CODE REVIEW

File: openassessment/templates/openassessmentblock/message/oa_message_no_team.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% load i18n %}
+{% spaceless %}
+<div id="openassessment__message__{{ xblock_id }}" class="message--warning openassessment__message message">
+    <h4 class="message__title">{% trans "Team membership required to view assignment" %} </h4>
```
Comment (Andytr1) [inline openassessment/templates/openassessmentblock/message/oa_message_no_team.html]: nit: would "Team membership **is** required..." be a better wording

File: openassessment/xblock/team_mixin.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        return bool(team)
+
+    def should_hide_team_ora(self):
+        """
+        Hide the ORA when the user is not on a team for a team ORA.
+        Ignore and return dummy data if we are course staff or in studio preview, since
+        we will want to see how the page will look to students
+        """
+        return (
+            (not self.has_team()) and not
```
Comment (iloveagent57) [inline openassessment/xblock/team_mixin.py]: Nit: putting the second `not` on the next line would make it more clear that it applies to the disjunction on the next line.

File: openassessment/xblock/test/test_team.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        ]
+
+    def get_team(self, user, course_id):  # pylint: disable=unused-argument
+        if self.has_team:
+            return self.team
+        else:
+            return None
+
+
+class MockBlock(TeamMixin):
```
Comment (iloveagent57) [inline openassessment/xblock/test/test_team.py]: I like this.

File: openassessment/xblock/test/test_team.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            }
+        )
+
+    @ddt.unpack
+    @ddt.data(
+        (False, False, False, True),  # Student in LMS with no team
+        (True, False, False, False),  # Student in LMS with a team
+        (False, False, True, False),  # Studio preview
+        (False, True, False, False),  # Course staff in LMS
+        (False, True, True, False),   # Course Staff in studio preview
```
Comment (iloveagent57) [inline openassessment/xblock/test/test_team.py]: Nice

File: openassessment/xblock/team_mixin.py (hunk 2 above)
Comment (jansenk) [reply openassessment/xblock/team_mixin.py]: it's a pep8 violation to both have a newline before `and` and after `and`, so I just put everything on one line

**Pull request closed without merging**